react-form-controls/src/NewsletterForm.jsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { input: '' };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(e) {
+    this.setState({ input: e.target.value });
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    console.log(this.state);
+  }
+
+  render() {
+    console.log(this.state.input);
+    return (
+      <form onSubmit={this.handleSubmit} action="">
+        <label htmlFor="email">Email</label>
+        <input onChange={this.handleChange} type="text" name="email" id="email" />
```

Comment:
The `value` attribute of the input needs to be set from the value in state `value={this.state.input}`

Reply:
@scbowler I updated and added value to input

Reply:
@byu014 Looks good! Thanks for updating, I didn't mean to approve... Approved now for reals ✅